src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,379 @@
+ * inertial coordinates and \f$C\leq 1\f$ is a free parameter (see e.g.
+ * \cite HesthavenWarburton, section 7.2).
+ */
+template <size_t Dim, typename FluxesComputerTag, typename FieldTagsList,
+          typename AuxiliaryFieldTagsList,
+          typename FluxesComputer = db::item_type<FluxesComputerTag>,
+          typename FluxesArgs = typename FluxesComputer::argument_tags>
+struct FirstOrderInternalPenalty;
+
+template <size_t Dim, typename FluxesComputerTag, typename... FieldTags,
+          typename... AuxiliaryFieldTags, typename FluxesComputer,
+          typename... FluxesArgs>
+struct FirstOrderInternalPenalty<Dim, FluxesComputerTag,
+                                 tmpl::list<FieldTags...>,
+                                 tmpl::list<AuxiliaryFieldTags...>,
+                                 FluxesComputer, tmpl::list<FluxesArgs...>> {
+ private:
+  using fluxes_computer_tag = FluxesComputerTag;
+
+  template <typename Tag>
+  struct NormalDotDivFlux : db::PrefixTag, db::SimpleTag {
+    using tag = Tag;
+    using type = TensorMetafunctions::remove_first_index<typename Tag::type>;
+  };
+
+  template <typename Tag>
+  struct NormalDotNormalDotFlux : db::PrefixTag, db::SimpleTag {
+    using tag = Tag;
+    using type = TensorMetafunctions::remove_first_index<typename Tag::type>;
+  };
+
+ public:
+  struct PenaltyParameter {
+    using type = double;
+    // Currently this is used as the full prefactor to the penalty term. When it
+    // becomes possible to communicate non-tensors (the element size $h$ and
+    // the number of collocation points $p$ on either side of the mortar), this
+    // should be changed to be just the parameter multiplying
+    // $\frac{N_\mathrm{points}^2}{h}$. Currently, the `packaged_data` is a
+    // `Variables` which can only hold tensors.
+    static constexpr OptionString help = {
+        "The prefactor to the penalty term of the flux."};
+  };
+  using options = tmpl::list<PenaltyParameter>;
+  static constexpr OptionString help = {
+      "The internal penalty flux for elliptic systems."};
+  static std::string name() noexcept { return "InternalPenalty"; }
+
+  FirstOrderInternalPenalty() = default;
+  explicit FirstOrderInternalPenalty(double penalty_parameter)
+      : penalty_parameter_(penalty_parameter) {}
+
+  // clang-tidy: non-const reference
+  void pup(PUP::er& p) noexcept { p | penalty_parameter_; }  // NOLINT
+
+  // These tags are sliced to the interface of the element and passed to
+  // `package_data` to provide the data needed to compute the numerical fluxes.
+  using argument_tags =
+      tmpl::list<::Tags::NormalDotFlux<AuxiliaryFieldTags>...,
+                 ::Tags::div<::Tags::Flux<AuxiliaryFieldTags, tmpl::size_t<Dim>,
+                                          Frame::Inertial>>...,
+                 fluxes_computer_tag, FluxesArgs...,
+                 ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim>>>;
+  using volume_tags = tmpl::list<fluxes_computer_tag>;
```

Comment:
What does volume_tags do here?

Reply:
It works like interface compute items (as of #1373): the `volume_tags` are taken from the DataBox directly instead of from the interface.